contracts/Clearable.sol

```diff
@@ -0,0 +1,159 @@
+        emit ClearableTransferExecuted(msg.sender, operationId);
+
+        return true;
+    }
+
+    function rejectClearableTransfer(string calldata operationId, string calldata reason) external returns (bool) {
+        ClearableTransfer storage newClearableTransfer = clearableTransfers[operationId.toHash()];
+        require (msg.sender == clearableAgent || newClearableTransfer.to == msg.sender, "Can only be rejected by the agent or the payee");
+        require (newClearableTransfer.status == ClearableTransferStatusCode.Ordered || newClearableTransfer.status == ClearableTransferStatusCode.InProcess, "A transfer can only be rejected in status Ordered or InProcess");
+        super._releaseHold(operationId);
+        newClearableTransfer.status = ClearableTransferStatusCode.Rejected;
+        emit ClearableTransferRejected(msg.sender, operationId, reason);
+
+        return true;
+    }
+
+    function retrieveClearableTransferData(string calldata operationId) external view returns (address from, address to, uint256 value, ClearableTransferStatusCode status) {
+        ClearableTransfer storage newClearableTransfer = clearableTransfers[operationId.toHash()];
+        return(
+            newClearableTransfer.from,
+            newClearableTransfer.to,
+            newClearableTransfer.value,
+            newClearableTransfer.status
+            );
+    }
+
+    function authorizeClearableTransferOperator(address operator) external returns (bool) {
+        require (operators[msg.sender][operator] == false, "The operator is already authorized");
+
+        operators[msg.sender][operator] = true;
+        emit AuthorizedClearableTransferOperator(operator, msg.sender);
+        return true;
+    }
+
+    function revokeClearableTransferOperator(address operator) external returns (bool) {
+        require (operators[msg.sender][operator] == true, "The operator is already not authorized");
+
+        operators[msg.sender][operator] = false;
+        emit RevokedClearableTransferOperator(operator, msg.sender);
+        return true;
+    }
+
+    function isClearableTransferOperatorFor(address operator, address from) external view returns (bool) {
+        return operators[from][operator];
+    }
+
+    function defineClearableAgent (address newClearableAgent) onlyOwner external  returns (bool) {
+        clearableAgent = newClearableAgent;
+        return true;
+    }
+
+    function isClearableAgent(address agent) external returns (bool) {
+        return agent == clearableAgent;
+    }
+
+    function _orderTransfer(string memory operationId, address orderer, address from, address to, uint256 value) internal returns (bool) {
+        super._hold(
+            operationId,
+            orderer,
+            from,
+            to,
+            clearableAgent,
+            value,
+            0
+        );
+        clearableTransfers[operationId.toHash()].orderer = orderer;
```

Comment:
It costs less gas to get the struct only once, like in the other functions with 
`ClearableTransfer storage newClearableTransfer = clearableTransfers[operationId.toHash()];`

Every time to you get values from the storage, like with `clearableTransfers[operationId.toHash()]` it costs much more gas, than doing it once and only accessing it from the memory.

Reply:
Great! thanks!